src/bootstrap/src/core/builder/cargo.rs

```diff
@@ -563,6 +563,12 @@ impl Builder<'_> {
             println!("using sysroot {sysroot_str}");
         }
 
+        if self.config.codegen_backends(target).first().map(|b| b.as_str()) == Some("gcc") {
+            let gcc_sysroot = self.config.libgccjit_root();
+            let gcc_sysroot = gcc_sysroot.as_os_str().to_str().expect("sysroot should be UTF-8");
+            cargo.env("LD_LIBRARY_PATH", gcc_sysroot);
```

Comment:
How is it done for cg_llvm?
Couldn't we reuse what is done there?

Reply:
That works for build, but not for dist, we do not want to dist libgccjit.so by default with rustup (at least not yet). If build did this automatically, we would have to explicitly skip it when creating the dist archives (normally we mostly copy all .so files to them, I think).

Reply:
I don't think it's very explicit in this case: https://github.com/rust-lang/rust/blob/e9c9b03652154c15b2bfc9f655d54e20f144a2d3/src/bootstrap/src/core/build_steps/dist.rs#L451

To clarify, the `cg_gcc` rustup component should include both `cg_gcc.so` and `libgccjit.so`. But the normal rustc component shouldn't contain `libgccjit.so`.